clients/rest-client-generator/src/test/java/uk/gov/justice/services/clients/rest/generator/RestClientGenerator_CodeStructureTest.java

```diff
@@ -0,0 +1,191 @@
+import static uk.gov.justice.services.adapters.test.utils.builder.RamlBuilder.raml;
+import static uk.gov.justice.services.adapters.test.utils.builder.RamlBuilder.restRamlWithDefaults;
+import static uk.gov.justice.services.adapters.test.utils.builder.ResourceBuilder.resource;
+import static uk.gov.justice.services.adapters.test.utils.config.GeneratorConfigUtil.configurationWithBasePackage;
+import static uk.gov.justice.services.adapters.test.utils.reflection.ReflectionUtil.firstMethodOf;
+import static uk.gov.justice.services.adapters.test.utils.reflection.ReflectionUtil.methodsOf;
+
+public class RestClientGenerator_CodeStructureTest {
+
+    private static final String BASE_PACKAGE = "org.raml.test";
+    private static final Map<String, String> NOT_USED_GENERATOR_PROPERTIES = ImmutableMap.of("serviceComponent", "QUERY_CONTROLLER");
+
+    private RestClientGenerator restClientGenerator;
+
+    private JavaCompilerUtil compiler;
+
+    @Rule
+    public TemporaryFolder outputFolder = new TemporaryFolder();
+
+    @Before
+    public void before() {
+        restClientGenerator = new RestClientGenerator();
+        compiler = new JavaCompilerUtil(outputFolder.getRoot(), outputFolder.getRoot());
+    }
+
+    @Test
+    public void shouldGenerateClassWithAnnotations() throws Exception {
+
+        restClientGenerator.run(
+                raml()
+                        .withBaseUri("http://localhost:8080/warname/query/api/rest/service")
+                        .with(resource("/some/path/{recipeId}")
+                                .with(action(POST, "application/vnd.cakeshop.commands.add-recipe+json")
+                                        .withQueryParameters(queryParameterOf("recipename", true), queryParameterOf("topingredient", false))
+                                        .withActionWithResponseTypes("application/vnd.cakeshop.commands.cmd1+json"))
+                        ).build()
+                , configurationWithBasePackage(BASE_PACKAGE, outputFolder, ImmutableMap.of("serviceComponent", "QUERY_API")));
+
+
+
+        Class<?> applicationClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteServiceQueryApi");
+
+        assertThat(applicationClass.getCanonicalName(), is("org.raml.test.RemoteServiceQueryApi"));
+        assertThat(applicationClass.getAnnotation(Remote.class), not(nullValue()));
+        assertThat(applicationClass.getAnnotation(ServiceComponent.class), not(nullValue()));
+        assertThat(applicationClass.getAnnotation(ServiceComponent.class).value().toString(), is("QUERY_API"));
+        assertBaseUriField(applicationClass.getDeclaredField("BASE_URI"));
+        assertRestClientField(applicationClass.getDeclaredField("restClientProcessor"));
+        assertRestClientHelperField(applicationClass.getDeclaredField("restClientHelper"));
+    }
+
+    @Test
+    public void shouldGenerateClassWithQueryControllerAnnotation() throws Exception {
+        restClientGenerator.run(
```

Comment:
It is often subtly different for each test, so I think its best to keep it inline so it is clear what each test is doing